File: algorithms/algorithms-cplusplus/leetcode/Question_0605_Can_Place_Flowers.cc
#include <vector>

using namespace std;

class Solution {
 public:
  bool canPlaceFlowers(vector<int>& flowerbed, int n) {
    if (n <= 0) {
      return true;
    }
    int size = flowerbed.size();
    for (int i = 0; i < size; i++) {
      if (flowerbed[i] == 1) {
        continue;
      }
      if ((i == 0 || flowerbed[i - 1] == 0) && (i == size - 1 || flowerbed[i + 1] == 0)) {
        n--;
        flowerbed[i] = 1;
      }
      if (n == 0) {
        return true;
      }
    }
    return false;
  }
};

File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0605_Can_Place_Flowers.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0605_Can_Place_Flowers {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        if (flowerbed == null || n == 0) {
            return true;
        }
        for (int i = 0; i < flowerbed.length; i++) {
            if (flowerbed[i] == 1) {
                continue;
            }
            if ((i == 0 || flowerbed[i - 1] == 0) &&
                    (i == flowerbed.length - 1 || flowerbed[i + 1] == 0)) {
                n--;
                flowerbed[i] = 1;
            }
            if (n == 0) {
                return true;
            }
        }
        return false;
    }
}
